pkg/asset/installconfig/aws/validation.go

```diff
@@ -298,8 +300,17 @@ func validateServiceEndpoints(fldPath *field.Path, region string, services []aws
 		}
 		conn.Close()
 	}
-	if _, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
-		return allErrs
+
+	if partition, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
+		if _, ok := partition.Regions()[region]; ok {
+			return allErrs
+		} else if len(services) == 0 {
+			err := validateRegion(region)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), region, "region not found"))
+				return allErrs
+			}
+		}
```

Comment:
When the partition is found, this needs to *always* return so that the resolver code is not called. Otherwise, valid regions that are not known by the AWS SDK will fail.

This also means that we should not be checking whether the user provided services. Instead, the `validateRegion` function needs to take as a parameter the services so that the user-provided endpoint for ec2 can be used if provided. This leaves a gap where the user could provide an invalid region and a valid ec2 endpoint, but I think I can live with that.
```suggestion
		if _, ok := partition.Regions()[region]; !ok {
			err := validateRegion(region, serviceEndpoints)
			if err != nil {
				allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), region, "region not found"))
			}
		}
		return allErrs
```